src/Uno.UI/UI/Xaml/UIElement.RoutedEvents.cs

```diff
@@ -740,7 +740,15 @@ internal bool RaiseEvent(RoutedEvent routedEvent, RoutedEventArgs args, Bubbling
 			}
 
 			UIElement parent = null;
-			if (this is not PopupPanel)
+			if (this is Popup popup)
+			{
+				// When a popup is clicked, the event propagation starts at
+				// the popup itself, and not it's (logical) child, so we need to
+				// handle this case explicitly. Note that the logical child isn't
+				// actually a child, so the event won't propagate back to the popup.
+				parent = popup.Child;
```

Comment:
Does that ,eam the popup get the pointer event before its child? Is this the case on Windows?

Reply:
Definitely want @dr1rrb to review this :)

Reply:
As far as I've tested, MenuFlyoutPresenter seems to be the only element to receive a KeyDown. This code has a problem if someone subscribes to Popup, it should be fixed now.